---
title: "Missing Feature Name"
layout: single
permalink: /docs/rules/missing-feature-name/
---

Your `Feature` should have a name that describes what the feature is doing. This makes it clear to the reader what the feature is doing and why it is important.

## Bad

```gherkin
Feature:
```

## Good

```gherkin
Feature: logging in
```
