---
title: "Missing Scenario Name"
layout: single
permalink: /docs/rules/missing-scenario-name/
---

Your `Scenario` and `Scenario Outline` should have a name that describes what the feature is doing. This makes it clear to the reader what the feature is doing and why it is important.

## Bad

```gherkin
Scenario:
```

## Good

```gherkin
Scenario: logging in
```
