# Missing Feature Name

Your `Feature` should have a name that describes what the feature is doing. This makes it clear to the reader what the feature is doing and why it is important.

## Bad

```gherkin
Feature:
```

## Good

```gherkin
Feature: logging in
```